pkg/stub/handler.go

```diff
@@ -128,17 +129,24 @@ func (h *Handler) StatusUpdate(condition v1alpha1.SamplesResourceConditionType,
 			cm.Data[installtypekey] = string(srcfg.Spec.InstallType)
 		}
 		if len(srcfg.Spec.Architectures) == 0 {
-			cm.Data[x86] = x86
+			cm.Data[v1alpha1.X86Architecture] = v1alpha1.X86Architecture
 		} else {
 			for _, arch := range srcfg.Spec.Architectures {
 				switch arch {
-				case x86:
-					cm.Data[x86] = x86
-				case ppc:
-					cm.Data[ppc] = ppc
+				case v1alpha1.X86Architecture:
+					cm.Data[v1alpha1.X86Architecture] = v1alpha1.X86Architecture
+				case v1alpha1.PPCArchitecture:
+					cm.Data[v1alpha1.PPCArchitecture] = v1alpha1.PPCArchitecture
 				}
 			}
 		}
+		or := metav1.OwnerReference{
+			APIVersion: "samplesoperator.config.openshift.io/v1alpha1",
+			Kind:       "SamplesResource",
+			Name:       v1alpha1.SamplesResourceName,
+			UID:        srcfg.UID,
+		}
+		cm.OwnerReferences = append(cm.OwnerReferences, or)
```

Comment:
so one caveat about relying on ownerreferences that might push us to handle the deletion explicitly in the finalizer instead:

Depending how the samplesresource is deleted(the deleteoptions passed during the delete request), the configmap may or may not be deleted before the samplesresource is deleted.  So if you really want to ensure the old configmap is gone before the samplesresource gets deleted/recreated(which i think we need to), I think you need to handle it explicitly in your finalizer instead.

Sorry I didn't think about that when i suggested using GC.